### DISCUSSION TITLE
Inquiry Regarding DeepSpeed Trademark Use for Hardware Vendor Fork

### DISCUSSION THREAD

**Discussion by kevint324** [deepspeedai/DeepSpeed#5399] [Q&A] [answered]
Dear DeepSpeed Community,

I hope this message finds you well. I am reaching out from a hardware vendor perspective with a question concerning the use of the DeepSpeed trademark, a project that I understand is a registered trademark of Microsoft.

My team have recently forked the DeepSpeed project to develop and implement additional functionalities specifically tailored to hardware backend extensions. In doing so, we have ensured that all original licenses and declarations as stipulated by the DeepSpeed project have been retained in full, with our modifications solely focusing on these functional extensions.

As we prepare to distribute our modified version of DeepSpeed to our clients, we seek clarification on the appropriate use of the DeepSpeed name within our project. Specifically, we would like to understand:

Whether it is necessary for us to alter our project's name to avoid any potential trademark issues with the original DeepSpeed trademark owned by Microsoft.
If a name change is indeed required, could you advise us on whether it is permissible to include any variant of the “DeepSpeed” string within our new project name, or should we avoid its use entirely to ensure compliance with trademark policies?
We aim to respect all legal and ethical considerations surrounding the DeepSpeed trademark and wish to ensure our project adheres to the highest standards of compliance. Your guidance on this matter would be greatly appreciated.

Thank you for your time and assistance. We look forward to your response and any advice you could provide on ensuring our project remains within the bounds of trademark use regulations.

Warm regards,

**Comment by martincai** [comment]
Hi Kevin, thank you for contacting us on this topic.  Appreciated.
Please give me a couple of days to get back to you.

**Comment by martincai** [comment, marked as answer]
Hi Kevin,

Yes, Microsoft owns registered rights in the DeepSpeed trademark. You should not use "DeepSpeed" in the name of your business or product. Please find an alternative mark. However, you may use "DeepSpeed" descriptively to communicate the relationship of your product with the DeepSpeed open-source project. For example, to describe the origin of the code and to reproduce the contents of the Notice file.

I hope this answers your question.

**Reply by kevint324** [reply]
Hi Martin

We were hestitating between DeepSpeed-XXX and XXXSpeed.
Now it seems clear that we should go for the second one.

Thanks for the clarification. Really appreciate it.

Cheers
Kevin
